lib/std/crypto/aes_cmac.zig

```diff
@@ -0,0 +1,171 @@
+/// RFC 4493 - The AES-CMAC Algorithm
+/// https://www.rfc-editor.org/rfc/rfc4493
+const std = @import("std");
+const Aes128 = std.crypto.core.aes.Aes128;
+const print = std.debug.print;
+
+pub fn AesCmac() type {
+    return struct {
+        const Self = @This();
+        var k1: [16]u8 = undefined;
+        var k2: [16]u8 = undefined;
+        var ctx: std.crypto.core.aes.AesEncryptCtx(Aes128) = undefined;
```

Comment:
These look like they're supposed to be fields?

Reply:
k1 and k2 are buffers for the sub keys and will are calculated at runtime, see Figure 2.2 in RFC 4493.

Reply:
global variable:
```zig
var k1: [16]u8 = undefined;
```

struct field:
```zig
k1: [16]u8 = undefined,
```

also: recommended to not default-initialize struct fields to `undefined`, especially in crypto code.

Reply:
They are always initialized in the Init area:
```
            k1 = [_]u8{0} ** 16;
            k2 = [_]u8{0} ** 16;
```

Reply:
But you are right, I can move them to the `generate` function.

Reply:
please read @Vexu and my comments again, carefully, and acknowledge what we are trying to tell you.

Reply:
I have changed that and it should be ok now.